openfecwebapp/views.py

```diff
@@ -0,0 +1,70 @@
+from openfecwebapp.api_caller import load_totals
+from flask import render_template
+from openfecwebapp.data_mappings import (type_map, 
+map_candidate_table_values, map_committee_table_values, 
+map_candidate_page_values, map_totals, committee_type_map,
+generate_pagination_values)
+from werkzeug.exceptions import abort
+
+def render_search_results(results, query):
+    candidates = []
+    committees = []
+
+    if results.get('candidates') and results['candidates'].get(
+        'results'):
+        for c in results['candidates']['results']:
+            candidates.append(map_candidate_table_values(c))
+
+    if results.get('committees') and results['committees'].get(
+        'results'):
+        for c in results['committees']['results']:
+            committees.append(map_committee_table_values(c))
+
+    # if true will show "no results" message
+    no_results = True if not len(candidates) and not len(committees) else False
+
+    return render_template('search-results.html', candidates=candidates,
+        committees=committees, query=query, no_results=no_results)
+
+# loads browse tabular views
+def render_table(data_type, results, params, url):
+    # if we didn't get data back from the API
+    if len(results) is 0:
+        abort(500)
+
+    results_table = {}
+    results_table[data_type] = []
+    heading = "Browse " + data_type
+
+    results_table['pagination'] = generate_pagination_values(results, params, url, data_type)
+
+    for r in results['results']:
+        results_table[data_type].append(type_map[data_type](r))
+
+    return render_template(data_type + '.html', **results_table)
+
+def render_page(data_type, c_data):
+    # not handling error at api module because sometimes its ok to 
+    # not get data back - like with search results
+    if len(c_data['results']) is 0:
```

Comment:
```python
c_data['results']
``` will throw an exception if "results" is not present in the dict. Are you thinking of `get`?